error: expected `,`
 --> $DIR/routes.rs:4:23
  |
4 |     let _ = routes![a b]; //~ ERROR expected `,`
  |                       ^

error: expected identifier
 --> $DIR/routes.rs:6:24
  |
6 |     let _ = routes![a::, ]; //~ ERROR expected identifier
  |                        ^

error: unexpected end of input, expected identifier
 --> $DIR/routes.rs:7:13
  |
7 |     let _ = routes![a::]; //~ ERROR expected identifier
  |             ^^^^^^^^^^^^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
