error: expected `,`
 --> $DIR/catchers.rs:4:25
  |
4 |     let _ = catchers![a b]; //~ ERROR expected
  |                         ^

error: expected identifier
 --> $DIR/catchers.rs:6:26
  |
6 |     let _ = catchers![a::, ]; //~ ERROR expected identifier
  |                          ^

error: unexpected end of input, expected identifier
 --> $DIR/catchers.rs:7:13
  |
7 |     let _ = catchers![a::]; //~ ERROR expected identifier
  |             ^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 3 previous errors

