error: invalid path URI: expected token '/' but found 'a' at index 0
 --> $DIR/route-path-bad-syntax.rs:5:8
  |
5 | #[get("a")] //~ ERROR invalid path URI
  |        ^^
  |
  = help: expected path in origin form: "/path/<param>"

error: invalid path URI: expected token '/' but none was found at index 0
 --> $DIR/route-path-bad-syntax.rs:9:8
  |
9 | #[get("")] //~ ERROR invalid path URI
  |        ^
  |
  = help: expected path in origin form: "/path/<param>"

error: invalid path URI: expected token '/' but found 'a' at index 0
  --> $DIR/route-path-bad-syntax.rs:13:8
   |
13 | #[get("a/b/c")] //~ ERROR invalid path URI
   |        ^^^^^^
   |
   = help: expected path in origin form: "/path/<param>"

error: paths cannot contain empty segments
  --> $DIR/route-path-bad-syntax.rs:17:7
   |
17 | #[get("/a///b")] //~ ERROR empty segments
   |       ^^^^^^^^
   |
   = note: expected '/a/b', found '/a///b'

error: query cannot contain empty segments
  --> $DIR/route-path-bad-syntax.rs:21:10
   |
21 | #[get("/?bat&&")] //~ ERROR empty segments
   |          ^^^^^

error: query cannot contain empty segments
  --> $DIR/route-path-bad-syntax.rs:24:10
   |
24 | #[get("/?bat&&")] //~ ERROR empty segments
   |          ^^^^^

error: paths cannot contain empty segments
  --> $DIR/route-path-bad-syntax.rs:27:7
   |
27 | #[get("/a/b//")] //~ ERROR empty segments
   |       ^^^^^^^^
   |
   = note: expected '/a/b', found '/a/b//'

error: invalid path URI: expected EOF but found '#' at index 3
  --> $DIR/route-path-bad-syntax.rs:33:11
   |
33 | #[get("/!@#$%^&*()")] //~ ERROR invalid path URI
   |           ^^^^^^^^^
   |
   = help: expected path in origin form: "/path/<param>"

error: component contains invalid URI characters
  --> $DIR/route-path-bad-syntax.rs:37:9
   |
37 | #[get("/a%20b")] //~ ERROR invalid URI characters
   |         ^^^^^
   |
   = note: components cannot contain reserved characters
   = help: reserved characters include: '%', '+', '&', etc.

error: component contains invalid URI characters
  --> $DIR/route-path-bad-syntax.rs:42:11
   |
42 | #[get("/a?a%20b")] //~ ERROR invalid URI characters
   |           ^^^^^
   |
   = note: components cannot contain reserved characters
   = help: reserved characters include: '%', '+', '&', etc.

error: component contains invalid URI characters
  --> $DIR/route-path-bad-syntax.rs:47:11
   |
47 | #[get("/a?a+b")] //~ ERROR invalid URI characters
   |           ^^^
   |
   = note: components cannot contain reserved characters
   = help: reserved characters include: '%', '+', '&', etc.

error: unused dynamic parameter
  --> $DIR/route-path-bad-syntax.rs:54:9
   |
54 | #[get("/<name>")] //~ ERROR unused dynamic parameter
   |         ^^^^^^
   |
note: expected argument named `name` here
  --> $DIR/route-path-bad-syntax.rs:55:7
   |
55 | fn h0(_name: usize) {} //~ NOTE expected argument named `name` here
   |       ^^^^^^^^^^^^

error: unused dynamic parameter
  --> $DIR/route-path-bad-syntax.rs:57:11
   |
57 | #[get("/a?<r>")] //~ ERROR unused dynamic parameter
   |           ^^^
   |
note: expected argument named `r` here
  --> $DIR/route-path-bad-syntax.rs:58:1
   |
58 | fn h1() {} //~ NOTE expected argument named `r` here
   | ^^^^^^^^^^

error: unused dynamic parameter
  --> $DIR/route-path-bad-syntax.rs:60:22
   |
60 | #[post("/a", data = "<test>")] //~ ERROR unused dynamic parameter
   |                      ^^^^^^
   |
note: expected argument named `test` here
  --> $DIR/route-path-bad-syntax.rs:61:1
   |
61 | fn h2() {} //~ NOTE expected argument named `test` here
   | ^^^^^^^^^^

error: unused dynamic parameter
  --> $DIR/route-path-bad-syntax.rs:63:9
   |
63 | #[get("/<_r>")] //~ ERROR unused dynamic parameter
   |         ^^^^
   |
note: expected argument named `_r` here
  --> $DIR/route-path-bad-syntax.rs:64:1
   |
64 | fn h3() {} //~ NOTE expected argument named `_r` here
   | ^^^^^^^^^^

error: unused dynamic parameter
  --> $DIR/route-path-bad-syntax.rs:66:9
   |
66 | #[get("/<_r>/<b>")] //~ ERROR unused dynamic parameter
   |         ^^^^
   |
note: expected argument named `_r` here
  --> $DIR/route-path-bad-syntax.rs:68:1
   |
68 | fn h4() {} //~ NOTE expected argument named `_r` here
   | ^^^^^^^^^^

error: unused dynamic parameter
  --> $DIR/route-path-bad-syntax.rs:66:14
   |
66 | #[get("/<_r>/<b>")] //~ ERROR unused dynamic parameter
   |              ^^^
   |
note: expected argument named `b` here
  --> $DIR/route-path-bad-syntax.rs:68:1
   |
68 | fn h4() {} //~ NOTE expected argument named `_r` here
   | ^^^^^^^^^^

error: `foo_.` is not a valid identifier
  --> $DIR/route-path-bad-syntax.rs:73:9
   |
73 | #[get("/<foo_.>")] //~ ERROR `foo_.` is not a valid identifier
   |         ^^^^^^^
   |
   = help: parameter names must be valid identifiers

error: `foo*` is not a valid identifier
  --> $DIR/route-path-bad-syntax.rs:77:9
   |
77 | #[get("/<foo*>")] //~ ERROR `foo*` is not a valid identifier
   |         ^^^^^^
   |
   = help: parameter names must be valid identifiers

error: `!` is not a valid identifier
  --> $DIR/route-path-bad-syntax.rs:81:9
   |
81 | #[get("/<!>")] //~ ERROR `!` is not a valid identifier
   |         ^^^
   |
   = help: parameter names must be valid identifiers

error: `name>:<id` is not a valid identifier
  --> $DIR/route-path-bad-syntax.rs:85:9
   |
85 | #[get("/<name>:<id>")] //~ ERROR `name>:<id` is not a valid identifier
   |         ^^^^^^^^^^^
   |
   = help: parameter names must be valid identifiers

error: malformed parameter
  --> $DIR/route-path-bad-syntax.rs:91:20
   |
91 | #[get("/", data = "foo")] //~ ERROR malformed parameter
   |                    ^^^
   |
   = help: parameter must be of the form '<param>'

error: malformed parameter
  --> $DIR/route-path-bad-syntax.rs:95:20
   |
95 | #[get("/", data = "<foo..>")] //~ ERROR malformed parameter
   |                    ^^^^^^^
   |
   = help: parameter must be of the form '<param>'

error: parameter is missing a closing bracket
  --> $DIR/route-path-bad-syntax.rs:99:20
   |
99 | #[get("/", data = "<foo")] //~ ERROR missing a closing bracket
   |                    ^^^^
   |
   = help: did you mean '<foo>'?

error: `test ` is not a valid identifier
   --> $DIR/route-path-bad-syntax.rs:103:20
    |
103 | #[get("/", data = "<test >")] //~ ERROR `test ` is not a valid identifier
    |                    ^^^^^^^
    |
    = help: parameter names must be valid identifiers

error: parameters must be named
   --> $DIR/route-path-bad-syntax.rs:109:9
    |
109 | #[get("/<_>")] //~ ERROR must be named
    |         ^^^
    |
    = help: use a name such as `_guard` or `_param`

error: parameter names cannot be empty
   --> $DIR/route-path-bad-syntax.rs:114:9
    |
114 | #[get("/<>")] //~ ERROR cannot be empty
    |         ^^

error: malformed parameter or identifier
   --> $DIR/route-path-bad-syntax.rs:117:9
    |
117 | #[get("/<id><")] //~ ERROR malformed parameter
    |         ^^^^^
    |
    = help: parameters must be of the form '<param>'
    = help: identifiers cannot contain '<' or '>'

error: malformed parameter or identifier
   --> $DIR/route-path-bad-syntax.rs:122:9
    |
122 | #[get("/<<<<id><")] //~ ERROR malformed parameter
    |         ^^^^^^^^
    |
    = help: parameters must be of the form '<param>'
    = help: identifiers cannot contain '<' or '>'

error: malformed parameter or identifier
   --> $DIR/route-path-bad-syntax.rs:127:9
    |
127 | #[get("/<>name><")] //~ ERROR malformed parameter
    |         ^^^^^^^^
    |
    = help: parameters must be of the form '<param>'
    = help: identifiers cannot contain '<' or '>'

error: aborting due to 30 previous errors

