warning: 'application/x-custom' is not a known media type
 --> $DIR/route-warnings.rs:7:21
  |
7 | #[get("/", format = "application/x-custom")] //~ WARNING not a known media type
  |                     ^^^^^^^^^^^^^^^^^^^^^^

warning: 'x-custom/plain' is not a known media type
  --> $DIR/route-warnings.rs:10:21
   |
10 | #[get("/", format = "x-custom/plain")] //~ WARNING not a known media type
   |                     ^^^^^^^^^^^^^^^^

warning: 'x-custom/x-custom' is not a known media type
  --> $DIR/route-warnings.rs:13:21
   |
13 | #[get("/", format = "x-custom/x-custom")] //~ WARNING not a known media type
   |                     ^^^^^^^^^^^^^^^^^^^

warning: `data` used with non-payload-supporting method
  --> $DIR/route-warnings.rs:18:12
   |
18 | #[get("/", data = "<_foo>")] //~ WARNING used with non-payload-supporting method
   |            ^^^^^^^^^^^^^^^
   |
note: 'GET' does not typically support payloads
  --> $DIR/route-warnings.rs:18:3
   |
18 | #[get("/", data = "<_foo>")] //~ WARNING used with non-payload-supporting method
   |   ^^^
   = note: this warning originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: `data` used with non-payload-supporting method
  --> $DIR/route-warnings.rs:21:13
   |
21 | #[head("/", data = "<_foo>")] //~ WARNING used with non-payload-supporting method
   |             ^^^^^^^^^^^^^^^
   |
note: 'HEAD' does not typically support payloads
  --> $DIR/route-warnings.rs:21:3
   |
21 | #[head("/", data = "<_foo>")] //~ WARNING used with non-payload-supporting method
   |   ^^^^
   = note: this warning originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: 5 warnings emitted

