error: named and unnamed parameters cannot be mixed
 --> $DIR/typed-uris-invalid-syntax.rs:7:18
  |
7 |     uri!(simple: id = 100, "Hello"); //~ ERROR named and unnamed
  |                  ^^^^^^^^^^^^^^^^^

error: named and unnamed parameters cannot be mixed
 --> $DIR/typed-uris-invalid-syntax.rs:8:18
  |
8 |     uri!(simple: "Hello", id = 100); //~ ERROR named and unnamed
  |                  ^^^^^^^^^^^^^^^^^

error: expected `:`
 --> $DIR/typed-uris-invalid-syntax.rs:9:16
  |
9 |     uri!(simple,); //~ ERROR expected `:`
  |                ^

error: expected argument list after `:`
  --> $DIR/typed-uris-invalid-syntax.rs:10:16
   |
10 |     uri!(simple:); //~ ERROR argument list
   |                ^

error: unexpected end of input: expected ',' followed by route path
  --> $DIR/typed-uris-invalid-syntax.rs:11:10
   |
11 |     uri!("/mount"); //~ ERROR route path
   |          ^^^^^^^^

error: unexpected end of input, expected identifier
  --> $DIR/typed-uris-invalid-syntax.rs:12:5
   |
12 |     uri!("/mount",); //~ ERROR expected identifier
   |     ^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: invalid mount point; mount points must be static, absolute URIs: `/example`
  --> $DIR/typed-uris-invalid-syntax.rs:13:10
   |
13 |     uri!("mount", simple); //~ invalid mount point
   |          ^^^^^^^

error: invalid mount point; mount points must be static, absolute URIs: `/example`
  --> $DIR/typed-uris-invalid-syntax.rs:14:10
   |
14 |     uri!("/mount/<id>", simple); //~ invalid mount point
   |          ^^^^^^^^^^^^^

error: unexpected end of input, call to `uri!` cannot be empty
  --> $DIR/typed-uris-invalid-syntax.rs:15:5
   |
15 |     uri!(); //~ unexpected end of input
   |     ^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected expression
  --> $DIR/typed-uris-invalid-syntax.rs:16:5
   |
16 |     uri!(simple: id = ); //~ expected expression
   |     ^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 10 previous errors

